polkadot/runtime/parachains/src/paras_inherent/weights.rs

```diff
@@ -156,6 +158,27 @@ pub fn signed_bitfield_weight<T: Config>(bitfield: &UncheckedSignedAvailabilityB
 	)
 }
 
+/// Worst case scenario is all candidates have been enacted
+/// and process a maximum number of messages.
+pub fn enact_candidates_max_weight<T: Config>(
+	bitfields: &UncheckedSignedAvailabilityBitfields,
+) -> Weight {
+	let config = configuration::ActiveConfig::<T>::get();
+	let max_ump_msgs = config.max_upward_message_num_per_candidate;
+	let max_hrmp_msgs = config.hrmp_max_message_num_per_candidate;
+	// No bitfields - no enacted candidates
+	let bitfield_size = bitfields.first().map(|b| b.unchecked_payload().0.len()).unwrap_or(0);
+	set_proof_size_to_tx_size(
```

Comment:
I think we should get rid of tracking the proof size in this pallet, the `BlockWeight` proof size limit is set to `u64::MAX` on the relaychain. It only makes sense for parachains to measure it.

Reply:
I agree, but it is orthogonal to this PR and I'd leave to for the later refactoring aka the long-term fix.

Reply:
@ordian @sandreim No, we actually use this to ensure we don't overfill the block with transactions.